packages/turf-boolean-disjoint/index.js

```diff
@@ -0,0 +1,165 @@
+    switch (geom1.type) {
+    case 'Point':
+        switch (geom2.type) {
+        case 'Point':
+            return !compareCoords(geom1.coordinates, geom2.coordinates);
+        case 'LineString':
+            return !isPointOnLine(geom2, geom1);
+        case 'Polygon':
+            return !inside(geom1, geom2);
+        }
+        /* istanbul ignore next */
+        break;
+    case 'LineString':
+        switch (geom2.type) {
+        case 'Point':
+            return !isPointOnLine(geom1, geom2);
+        case 'LineString':
+            return !isLineOnLine(geom1, geom2);
+        case 'Polygon':
+            return !isLineInPoly(geom2, geom1);
+        }
+        /* istanbul ignore next */
+        break;
+    case 'Polygon':
+        switch (geom2.type) {
+        case 'Point':
+            return !inside(geom2, geom1);
+        case 'LineString':
+            return !isLineInPoly(geom1, geom2);
+        case 'Polygon':
+            return !isPolyInPoly(geom2, geom1);
+        }
+    }
+}
+
+// http://stackoverflow.com/a/11908158/1979085
+function isPointOnLine(lineString, point) {
+    for (var i = 0; i < lineString.coordinates.length - 1; i++) {
+        if (isPointOnLineSegment(lineString.coordinates[i], lineString.coordinates[i + 1], point.coordinates)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function isLineOnLine(lineString1, lineString2) {
+    var doLinesIntersect = lineIntersect(lineString1, lineString2);
+    if (doLinesIntersect.features.length > 0) {
+        return true;
+    }
+    return false;
+}
+
+function isLineInPoly(polygon, lineString) {
+    var doLinesIntersect = lineIntersect(lineString, polyToLinestring(polygon));
+    if (doLinesIntersect.features.length > 0) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Is Polygon (geom1) in Polygon (geom2)
+ * Only takes into account outer rings
+ * See http://stackoverflow.com/a/4833823/1979085
+ *
+ * @private
+ * @param {Geometry|Feature<Polygon>} feature1 Polygon1
+ * @param {Geometry|Feature<Polygon>} feature2 Polygon2
+ * @returns {Boolean} true/false
+ */
+function isPolyInPoly(feature1, feature2) {
+    for (var i = 0; i < feature1.coordinates[0].length; i++) {
+        if (inside(feature1.coordinates[0][i], feature2)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function isPointOnLineSegment(LineSegmentStart, LineSegmentEnd, Point) {
+    var dxc = Point[0] - LineSegmentStart[0];
+    var dyc = Point[1] - LineSegmentStart[1];
+    var dxl = LineSegmentEnd[0] - LineSegmentStart[0];
+    var dyl = LineSegmentEnd[1] - LineSegmentStart[1];
+    var cross = dxc * dyl - dyc * dxl;
+    if (cross !== 0) {
+        return false;
+    }
+    if (Math.abs(dxl) >= Math.abs(dyl)) {
+        if (dxl > 0) {
+            return LineSegmentStart[0] <= Point[0] && Point[0] <= LineSegmentEnd[0];
+        } else {
+            return LineSegmentEnd[0] <= Point[0] && Point[0] <= LineSegmentStart[0];
```

Comment:
Where does this source code come from?

Lack of comments is hard to troubleshoot.